src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -26,16 +26,31 @@
 import org.slf4j.LoggerFactory;
 
 import org.apache.cassandra.cql3.Operator;
-import org.apache.cassandra.db.marshal.AbstractType;
-import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
 import org.apache.cassandra.index.sai.analyzer.AbstractAnalyzer;
-import org.apache.cassandra.index.sai.utils.TypeUtil;
+import org.apache.cassandra.index.sai.analyzer.NoOpAnalyzer;
+import org.apache.cassandra.index.sai.utils.IndexTermType;
 
-public class Expression
+public interface Expression
```

Comment:
The interface could use some JavaDoc explaining what an `Expression` is, and that we have two subclasses for indexed/not-indexed columns.